module/solutions/helper.js

```diff
@@ -142,4 +1012,2537 @@ module.exports = class SolutionsHelper {
+        );
+
+        if (!privateSolutionDetails.length > 0) {
+          // Data for program and solution creation
+          let programAndSolutionData = {
+            type: CONSTANTS.common.IMPROVEMENT_PROJECT,
+            subType: CONSTANTS.common.IMPROVEMENT_PROJECT,
+            isReusable: false,
+            solutionId: solutionData.solutionId,
+          };
+
+          if (solutionData.programId && solutionData.programId !== "") {
+            programAndSolutionData["programId"] = solutionData.programId;
+            programAndSolutionData["programName"] = solutionData.programName;
+          }
+          // create private program and solution
+          let solutionAndProgramCreation = await this.createProgramAndSolution(
+            userId,
+            programAndSolutionData,
+            userToken,
+            "true" // create duplicate solution
+          );
+
+          if (!solutionAndProgramCreation.success) {
+            throw {
+              status: HTTP_STATUS_CODE.bad_request.status,
+              message: CONSTANTS.apiResponses.SOLUTION_PROGRAMS_NOT_CREATED,
+            };
+          }
+          return resolve({
+            success: true,
+            result: solutionAndProgramCreation.result.solution._id,
+          });
+        } else {
+          return resolve({
+            success: true,
+            result: privateSolutionDetails[0]._id,
+          });
+        }
+      } catch (error) {
+        return resolve({
+          success: false,
+          status: error.status
+            ? error.status
+            : HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message,
+        });
+      }
+    });
+  }
+
+
+
+
+   /**
+   * Verify solution link
+   * @method
+   * @name verifyLink
+   * @param {String} solutionId - solution Id.
+   * @param {String} userId - user Id.
+   * @param {String} userToken - user token.
+   * @param {Boolean} createProject - create project.
+   * @param {Object} bodyData - Req Body.
+   * @param {Object} createPrivateSolutionIfNotTargeted - flag to create private program if user is non targeted
+   * @returns {Object} - Details of the solution.
+   */
+
+   static verifyLink(
+    link = "",
+    bodyData = {},
+    userId = "",
+    userToken = "",
+    createProject = true
+  ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let checkForTargetedSolution = await this.checkForTargetedSolution(
+          link,
+          bodyData,
+          userId,
+          userToken
+        );
+
+        if (
+          !checkForTargetedSolution ||
+          Object.keys(checkForTargetedSolution.result).length <= 0
+        ) {
+          return resolve(checkForTargetedSolution);
+        }
+
+        let solutionData = checkForTargetedSolution.result;
+        let isSolutionActive =
+          solutionData.status === CONSTANTS.common.INACTIVE ? false : true;
+        if (solutionData.type == CONSTANTS.common.OBSERVATION) {
```

Comment:
@Prajwal17Tunerlabs comment observation and survey condition add project check condition, if the type is not project send some error